data_diff/diff_tables.py

```diff
@@ -23,23 +23,36 @@ class TableSegment:
     database: Database
     table_path: DbPath
     key_column: str
-    extra_columns: Tuple[str, ...]
-    start: DbKey = None
-    end: DbKey = None
+    update_column: str = None
+    extra_columns: Tuple[str, ...] = ()
+    start_key: DbKey = None
+    end_key: DbKey = None
+    min_time: DbTime = None
+    max_time: DbTime = None
 
     _count: int = None
     _checksum: int = None
 
-    def _make_range_pred(self):
-        if self.start is not None:
-            yield Compare("<=", str(self.start), self.key_column)
-        if self.end is not None:
-            yield Compare("<", self.key_column, str(self.end))
+    def __post_init__(self):
+        if not self.update_column and (self.min_time or self.max_time):
+            raise ValueError("Error: min_time/max_time feature requires to specify 'update_column'")
+
+    def _make_key_range(self):
+        if self.start_key is not None:
+            yield Compare("<=", str(self.start_key), self.key_column)
+        if self.end_key is not None:
+            yield Compare("<", self.key_column, str(self.end_key))

```

Comment:
I think the key version will also benefit from a CLI arg 👍🏻 Thanks for changing its name and making it symmetric with time

Lots of people will probably instead of temporal just get the id of the last records when they did the last insertion, then do the check based on that

Reply:
We already have `-k`. Do you mean something different?